cmake/modules/extensions.cmake

```diff
@@ -2594,6 +2594,10 @@ Relative paths are only allowed with `-D${ARGV1}=<path>`")
         set(ZFILE_BOARD_IDENTIFIER ${BOARD_IDENTIFIER})
       endif()
     endif()
+    if(NOT DEFINED ZFILE_SUFFIX)
+      # Defaulting to system wide settings when SUFFIX is not given as argument
+      set(ZFILE_SUFFIX ${FILE_SUFFIX})
+    endif()
```

Comment:
this change impact behavior for cases where suffix should not be applied, such as board and shield dir files.
https://github.com/zephyrproject-rtos/zephyr/blob/9a0b8e7809b0e877d2be664dcc09f9ba117b8a19/cmake/modules/dts.cmake#L157
https://github.com/zephyrproject-rtos/zephyr/blob/9a0b8e7809b0e877d2be664dcc09f9ba117b8a19/cmake/modules/shields.cmake#L118
https://github.com/zephyrproject-rtos/zephyr/blob/9a0b8e7809b0e877d2be664dcc09f9ba117b8a19/cmake/modules/shields.cmake#L122

Reply:
This behavior was already present due to `FILE_SUFFIX` being inherited from cache scope.

If we don't want to apply the suffix in these places, then I can fix it now by dropping one of the commits and instead adding explicit `SUFFIX ${FILE_SUFFIX}` where it's needed for sure.

Reply:
I think we should avoid having FILE_SUFFIX applied on boards as that will open for misuse.
Instead hwmv2 provides board variant options.

Reply:
Agreed, files in `<board_dir>/<vendor>/<board>` should not have this applied, it should be applied to `<app>/<boards>/<board>.overlay` and `<app>/<boards>/<board>.dts` files

Reply:
Done.